### PR TITLE
top related backers in monthly report

### PR DESCRIPTION
**Pull request by xdamman**
The goal is to show the top backers (sponsors) that have backed related collectives last month

### CODE REVIEW

File: server/models/Group.js [hunk 1]
```diff
@@ -448,6 +449,15 @@ export default function(Sequelize, DataTypes) {
     },
 
     classMethods: {
+      createMany: (groups, defaultValues) => {
+        return Promise.map(groups, group => {
+          for (const attr in defaultValues) {
```
Comment (asood123) [inline server/models/Group.js]: I had the impression that for/while loops are generally discouraged, in favor of maps, forEach, etc. I don't know where I got that from though.

File: server/models/User.js [hunk 2]
```diff
@@ -336,10 +349,14 @@ export default (Sequelize, DataTypes) => {
 
       decryptId(encrypted) {
         return decrypt(encrypted);
+      },
+
+      getTopBackers(since, until, tags, limit) {
```
Comment (asood123) [inline server/models/User.js]: This is a classMethod, since it never references `this`.

File: test/user.model.test.js [hunk 3]
```diff
@@ -95,4 +92,102 @@ describe('user.models.test.js', () => {
...
+            expect(donations[0]).to.have.property("amount");
+            expect(donations[0]).to.have.property("currency");
+            expect(donations[0]).to.have.property("Group");
+            expect(donations[0].Group).to.have.property("name");
+          })
+      });
+    });    
+
+  });
+
```
Comment (asood123) [inline test/user.model.test.js]: Glad you added these tests.

File: templates/emails/group.monthlyreport.hbs [hunk 4]
```diff
@@ -98,18 +101,18 @@ We hope it's useful. Don't hesitate to send us your feedback by replying to this
...
+  <li><a href="https://opencollective.com/sfartandfilm?utm_source=email&utm_campaign=monthlyreport">San Francisco Art & Film for Teenagers</a>, is on a mission to make all the arts an important part of young people's lives. [<a href="https://opencollective.com/FreeRideToVote?utm_source=email&utm_campaign=monthlyreport#support">support that mission!</a>]</li> 
+</ul>
+
+<h2>Blog posts</h2>
+<li><a href="http://empathia.agency">Empathia Agency</a> (who works with us) is giving back to the open source community. They decided to add 1% to their bills to donate to the Open Source Collective + 1% coming from their margin. <a href="https://medium.empathia.agency/believe-in-giving-back-b451887d79cb#.5j4bysm00">Read more</a>.
+<li><a href="http://blog.idonethis.com/open-source-community/">How iDoneThis says thank you to the open source community</a>
 </ul>
 
 <h2>Changelog</h2>
```
Comment (asood123) [inline templates/emails/group.monthlyreport.hbs]: this is getting indented as a sub section of blog posts in sample emails.

File: templates/emails/group.monthlyreport.hbs [hunk 5]
```diff
@@ -81,14 +81,17 @@ We hope it's useful. Don't hesitate to send us your feedback by replying to this
 {{/each}}
 </table>
 
-<h2>Newest sponsors on OpenCollective</h2>
+{{#if topBackers}}
+<h2>Latest related backers</h2>
```
Comment (asood123) [inline templates/emails/group.monthlyreport.hbs]: Hard to parse. How about "Recent sponsors on Open Collective"? We don't need to tell them that we filtered to related collectives.

File: templates/emails/group.monthlyreport.hbs [hunk 6]
```diff
@@ -81,14 +81,17 @@ We hope it's useful. Don't hesitate to send us your feedback by replying to this
...
-<li><a href="https://opencollective.com/auth0">Auth0</a>: they are sponsoring open source projects related to authentication</li>
+{{#each topBackers}}
+  <li><a href="{{website}}">{{name}}</a> gave {{{donationsString}}}</li>
+{{/each}}
 </ul>
+{{/if}}
 
 {{#if group.related}}
-  <h2>Newest related collectives</h2>
+  <h2>Latest related collectives</h2>
```
Comment (asood123) [inline templates/emails/group.monthlyreport.hbs]: How about "Other related collectives"? Or "Check out other similar collectives"

File: templates/emails/group.monthlyreport.hbs [hunk 7]
```diff
@@ -81,14 +81,17 @@ We hope it's useful. Don't hesitate to send us your feedback by replying to this
...
   <ul>
   {{#each group.related}}
     <li><a href="{{publicUrl}}?utm_source=email&utm_campaign=monthlyreport">{{name}}</a>, on a mission to {{mission}} [<a href="{{publicUrl}}?utm_source=email&utm_campaign=monthlyreport#support">support that mission!</a>]</li>
```
Comment (asood123) [inline templates/emails/group.monthlyreport.hbs]: s/support that mission/support their mission. Or even shorter, s/support that mission/support them

**Comment by sedubois** [comment]
This PR breaks master: https://circleci.com/gh/OpenCollective/opencollective-api/2716

Don't know why circleci results aren't shown here?

File: test/user.model.test.js [hunk 8]
```diff
@@ -159,6 +159,7 @@ describe('user.models.test.js', () => {
 
     it('gets the top backers in open source', () => {
       return User.getTopBackers(new Date('2016-06-01'), new Date('2016-07-01'), ['open source'])
+        .tap(console.log)
```
Comment (xdamman) [inline test/user.model.test.js]: why do you want the tests to be more verbose?

File: test/user.model.test.js [hunk 9]
```diff
@@ -159,7 +159,6 @@ describe('user.models.test.js', () => {
 
     it('gets the top backers in open source', () => {
       return User.getTopBackers(new Date('2016-06-01'), new Date('2016-07-01'), ['open source'])
-        .tap(console.log)
```
Comment (xdamman) [inline test/user.model.test.js]: ah ok :)

**Comment by xdamman** [comment]
@sedubois because it has already been merged
